examples/token-stats/methods/guest/src/main.rs

```diff
@@ -26,18 +26,34 @@ fn main() {
     // Read the input from the guest environment.
     let input: EthEvmInput = env::read();
 
-    // Converts the input into a `EvmEnv` for execution. The `with_chain_spec` method is used
-    // to specify the chain configuration. It checks that the state matches the state root in the
-    // header provided in the input.
-    let env = input.into_env().with_chain_spec(&ETH_MAINNET_CHAIN_SPEC);
+    // Converts the input into a `EvmEnv` for execution.
+    let env1 = input.into_env().with_chain_spec(&ETH_MAINNET_CHAIN_SPEC);
 
     // Execute the view calls; it returns the result in the type generated by the `sol!` macro.
-    let contract = Contract::new(CONTRACT, &env);
+    let contract = Contract::new(CONTRACT, &env1);
     let utilization = contract
         .call_builder(&CometMainInterface::getUtilizationCall {})
         .call()
         ._0;
-    let supply_rate = contract
+    let supply_rate_1 = contract
+        .call_builder(&CometMainInterface::getSupplyRateCall { utilization })
+        .call()
+        ._0;
+
+    // Prepare the second `EvmEnv` for execution.
+    let input: EthEvmInput = env::read();
+    let env2 = input.into_env().with_chain_spec(&ETH_MAINNET_CHAIN_SPEC);
+
+    // Verify the first commitment.
+    SteelVerifier::new(&env2).verify(env1.commitment());
```

Comment:
It's not clear here how `env1` and `env2` and are related. I guess I'd assume `env2` needs to be a more recent block than `env1`, but the API does not make this obvious (I'm actually not even sure myself that this is correct).